src/main/scala/fs2/Process1.scala

```diff
@@ -172,6 +172,51 @@ object process1 {
     _ pull (go(0))
   }
 
+  /**
+    * Zip the elements of the input `Handle` with its next element wrapped into `Some`, and return the new `Handle`.
+    * The last element is zipped with `None`.
+    */
+  def zipWithNext[F[_], I]: Stream[F, I] => Stream[F, (I, Option[I])] = {
+    def go(last: I): Handle[F, I] => Pull[F, (I, Option[I]), Handle[F, I]] =
+      Pull.receiveOption {
+        case None =>
+          Pull.output1((last, None)) as Handle.empty
```

Comment:
No need for blankline here, and I'd move onto the same line as `case None` as it's a very short line and pretty easy to understand